ldm/invoke/generator/diffusers_pipeline.py

```diff
@@ -189,6 +188,25 @@ def __call__(self, *args: ParamType.args,
             raise AssertionError("why was that an empty generator?")
         return result
 
+
+@dataclass(frozen=True)
+class ConditioningData:
```

Comment:
oh. we are thinking the same way. i have done something similar on a local branch too. actually - i'm refactoring the conditioning provision system into a two-stage system. first, the prompt string and CFG scale gets passed to a `ConditioningSchedulerFactory` that returns a `ConditioningScheduler`. Then, every step the `ConditioningScaler` instance is queried to return what is basically your class here `ConditioningData` for that step. this will achieve a couple of things, including decoupling the scheduling info about cross-attention control that's bundled into `ExtraConditioningInfo` from the actual conditioning info itself, and second, it will enable more control over scheduling for conditioning params themselves, such as a way of setting a cosine-scaled CFG-scale (i.e. starts high and eases down to low as the steps progress) as well as non-cross-attention-controlled forms of switching up the prompt - (eg, run first 50% of steps with "a cat" and the second 50% of steps with "a fluffy cat with a shiny coat" or something)